src/mococrw/mac.h

```diff
@@ -145,4 +145,80 @@ class HMAC : public MessageAuthenticationCode {
     std::unique_ptr<Impl> _impl;
 };
 
+class CMAC : public MessageAuthenticationCode {
+public:
+    /**
+     * @brief Constructor
+     * @param hashFunction The hash function which shall be used
+     * @param key The key used for CMAC
+     */
+    CMAC(mococrw::openssl::CmacCipherTypes cipherType, const std::vector<uint8_t> &key);
+
+    /**
+     * @brief destructor
+     */
+    ~CMAC();
+
+    /**
+     * Adds the chunk of data to the hash. H(i-key || message1 [ || message2])
+     * For getting the CMAC invoke finish()
+     *
+     * @param message chunk of data to used for CMAC.
+     * @throws MoCOCrWException if this function is invoked after finish was called.
+     */
+    void update(const std::vector<uint8_t>& message) override;
+
+
+    /**
+     * @brief Finalize the CMAC.
+     *
+     * This function calculates the message authentication code.
+     * H(o-key || H(i-key || message [ || message]))
```

Comment:
This also applies here, this is how HMAC is computed.

Reply:
Done.